projects/plugins/crm/includes/ZeroBSCRM.Edit.Segment.php

```diff
@@ -163,100 +163,101 @@ function zeroBSCRM_html_addEditSegment($potentialID = -1)
-              echo json_encode( $jpcrm_mailpoet_status_list );
-            ?>
-            </script>
-
-    </div><?php
+						<?php
+						// simplify our external sources array
+						$external_source_array = array();
+						foreach ( $zbs->external_sources as $external_source_key => $external_source_info ) {
+							$external_source_array[] = array(
+								'key'  => $external_source_key,
+								'name' => $external_source_info[0],
+							);
+						}
+
+						// sort by name
+						usort(
+							$external_source_array,
+							function ( $a, $b ) {
+								return strcmp( $a['key'], $b['key'] );
+							}
+						);
+
+						// phpcs:disable WordPress.WP.I18n.TextDomainMismatch -- this is intentionally using the `mailpoet` text domain
+						$jpcrm_mailpoet_status_list = array(
+							array(
+								'key'  => 'subscribed',
+								'name' => __( 'Subscribed', 'mailpoet' ),
+							),
+							array(
+								'key'  => 'unconfirmed',
+								'name' => __( 'Unconfirmed', 'mailpoet' ),
+							),
+							array(
+								'key'  => 'unsubscribed',
+								'name' => __( 'Unsubscribed', 'mailpoet' ),
+							),
+							array(
+								'key'  => 'inactive',
+								'name' => __( 'Inactive', 'mailpoet' ),
+							),
+							array(
+								'key'  => 'bounced',
+								'name' => __( 'Bounced', 'mailpoet' ),
+							),
+						);
+						// phpcs:enable WordPress.WP.I18n.TextDomainMismatch
+						?>
+
+						<script type="text/javascript">
+						var zbsSegment = <?php echo wp_json_encode( $segment, JSON_UNESCAPED_SLASHES | JSON_HEX_TAG | JSON_HEX_AMP ); ?>;
+						var jpcrm_available_conditions = <?php echo wp_json_encode( $available_conditions, JSON_UNESCAPED_SLASHES | JSON_HEX_TAG | JSON_HEX_AMP ); ?>;
+						var jpcrm_available_conditions_by_category = <?php echo wp_json_encode( $available_conditions_by_category, JSON_UNESCAPED_SLASHES | JSON_HEX_TAG | JSON_HEX_AMP ); ?>;
+						var zbsAvailableConditionOperators = <?php echo wp_json_encode( $available_condition_operators, JSON_UNESCAPED_SLASHES | JSON_HEX_TAG | JSON_HEX_AMP ); ?>;
+						var jpcrm_available_contact_tags = <?php echo wp_json_encode( $available_tags_contacts, JSON_UNESCAPED_SLASHES | JSON_HEX_TAG | JSON_HEX_AMP ); ?>;
+						var jpcrm_available_transaction_tags = <?php echo wp_json_encode( $available_tags_transactions, JSON_UNESCAPED_SLASHES | JSON_HEX_TAG | JSON_HEX_AMP ); ?>;
+						var zbsAvailableStatuses = <?php echo wp_json_encode( $availableStatuses, JSON_UNESCAPED_SLASHES | JSON_HEX_TAG | JSON_HEX_AMP ); /* phpcs:ignore WordPress.NamingConventions.ValidVariableName.VariableNotSnakeCase */ ?>;
+						var zbsSegmentStemURL = '<?php echo jpcrm_esc_link( 'edit', -1, 'segment', true ); ?>';
+						var jpcrm_contact_stem_URL = '<?php echo jpcrm_esc_link( 'view', -1, 'contact', true ); ?>';
+						var zbsSegmentListURL = '<?php echo jpcrm_esc_link( $zbs->slugs['segments'] ); ?>';
+						var zbsSegmentSEC = '<?php echo esc_js( wp_create_nonce( 'zbs-ajax-nonce' ) ); ?>';
```

Comment:
`esc_js` is pretty much broken. I suspect it must have been written for versions of PHP that didn't have `json_encode` yet.
```suggestion
						var zbsSegmentSEC = <?php echo wp_json_encode( wp_create_nonce( 'zbs-ajax-nonce' ), JSON_UNESCAPED_SLASHES | JSON_HEX_TAG | JSON_HEX_AMP ); ?>;
```